scalalib/src/PublishModule.scala

```diff
@@ -106,12 +112,24 @@ trait PublishModule extends JavaModule { outer =>
 }
 
 object PublishModule extends ExternalModule {
-  case class PublishData(meta: Artifact, payload: Seq[(PathRef, String)])
 
+  case class PublishData(meta: Artifact, payload: Seq[(PathRef, String)])
   object PublishData{
     implicit def jsonify: upickle.default.ReadWriter[PublishData] = upickle.default.macroRW
   }
 
+  /** An extra resource artifact to publish.
+    * @param file The artifact file
+    * @param ivyCategory The ivy catogory (e.g. "jars", "zips")
+    * @params The file suffix including the file extension (e.g. "-with-deps.jar", "-dist.zip").
+    *        It will be appended to the artifact id to construct the full file name.
+    */
+  case class ExtraPublish(file: PathRef, ivyCategory: String, suffix: String)
```

Comment:
I've no idea, if `ivyCategory` is the right name for this kind of thing.

Reply:
Classifier, isn't that the right name? And shouldn't it be `@param suffix` there?

Reply:
I'm pretty sure it's not the classifier. It's something only needed for ivy repositories because of there layout. I think it's probably the type in a plural form?